math/gcd_of_n_numbers.cpp

```diff
@@ -1,41 +1,112 @@
 /**
  * @file
- * @brief This program aims at calculating the GCD of n numbers by division
- * method
+ * @brief This program aims at calculating the GCD of n numbers
  *
+ * @details
+ * The GCD of n numbers can be calculated by
+ * repeatedly calculating the GCDs of pairs of numbers
+ * i.e. gcd(a, b, c) = gcd(gcd(a, b), c)
+ * Euclidean algorithm helps calculate the GCD of each pair of numbers efficiently
+ * 
  * @see gcd_iterative_euclidean.cpp, gcd_recursive_euclidean.cpp
  */
-#include <iostream>
+#include <iostream>  /// for IO operations
+#include <cassert>  /// for assert
+#include <algorithm>  /// for std::abs
 
-/** Compute GCD using division algorithm
- *
- * @param[in] a array of integers to compute GCD for
- * @param[in] n number of integers in array `a`
+/**
+ * @namespace math
+ * @brief Maths algorithms
+ */
+namespace math {
+/**
+ * @namespace gcd_of_n_numbers
+ * @brief Compute GCD of numbers in an array
+ */
+namespace gcd_of_n_numbers {
+/**
+ * @brief Function to compute GCD of 2 numbers x and y
+ * @param x First number
+ * @param y Second number
+ * @return GCD of x and y via recursion
+ */
+int gcd_two(int x, int y) {
+    // base cases
+    if (y == 0) return x;
+    if (x == 0) return y;
+    return gcd_two(y, x % y);  // Euclidean method
+}
+/**
+ * @brief Function to check if all elements in array are 0
+ * @param a Array of numbers
+ * @param n Number of elements in array
+ * @return 'True' if all elements are 0
+ * @return 'False' if not all elements are 0
+ */
+bool check_all_zeros(int a[], size_t n) {
+    // Check for the undefined GCD cases
+    int zero_count = 0;
+    for (int i = 0; i < n; ++i) {
+        if (a[i] == 0) {
+            ++zero_count;
+        }
+    }
+
+    // return whether all elements in array are 0
+    return zero_count == n;
+}
+/** 
+ * @brief Main program to compute GCD using division algorithm
+ * @param a Array of integers to compute GCD for
+ * @param n number of integers in the array
+ * @return GCD of the numbers in the array
  */
-int gcd(int *a, int n) {
-    int j = 1;  // to access all elements of the array starting from 1
+int gcd(int a[], size_t n) {
```

Comment:
instead of using a fundamental array
an array from <array>
that is std::array<type>

this lets you remove the size parameter